Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -332,11 +357,21 @@ protected class ClassWriter : IClassWriter {
     protected override void EmitLiteralExpr(ConcreteSyntaxTree wr, LiteralExpr e) {
       if (e.Value is bool value) {
         wr.Write("{0}", value);
+
+      } else if (e is StringLiteralExpr) {
+        var str = (StringLiteralExpr)e;
+        TrStringLiteral(str, wr);
+      } else if (e.Value is BigInteger) {
+        var i = (BigInteger)e.Value;
+        wr.Write($"{i}");
       }
     }
 
     protected override void EmitStringLiteral(string str, bool isVerbatim, ConcreteSyntaxTree wr) {
-      throw new NotImplementedException();
+
+      if (!isVerbatim) {
+        wr.Write("\"{0}\"", str);
```

Comment:
Please escape characters accordingly (double quotes, perhaps newlines)

Reply:
I understand your concern, I just want to keep it related only to the test file  so nothing is added which is not used and thus less chance of errors.

Reply:
I understand your strategy and it does makes sense, especially in early stage.
However, consider this other point of view: this is intentionally implementing a function in a way that is obviously wrong (not incomplete, it returns a result even if this result is wrong) just to make that test pass. Although this might be fine in this particular case (which still bothers me) is in general a recipe for late errors that one will discover later and that will be more painful. On the other side, it's ok to implement something correctly that will tested later.

You might at least want to write the following:

`if(str.Contains("\"") || str.Contains("\\")) { throw new NotImplementedException(); }`

so that you are not caught by surprise later.

Reply:
just found out the flag isVerbatim is to check if the string has any escape characters, thanks for the comment I will add the `NotImplementedException()` in the else.

Reply:
I think that isVerbatim is a Dafny language that just checks if the flag "@" was in front of the original string, it does not assume anything about the content. Please double check.

Reply:
sure.

Reply:
Ok, thanks for double checking. Actually, it's my bad. Dafny records string escales in the string value directly, so there isn't anything you need to do, since the subset that Dafny supports is also supported by Python. You can consider this resolved. Thanks again and sorry for the trouble.

Reply:
It's always better to recheck, thanks for your input.